pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -84,11 +84,12 @@ func alterChangefeedPlanHook(
 	}
 
 	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) error {
-		typedExpr, err := alterChangefeedStmt.Jobs.TypeCheck(ctx, p.SemaCtx(), types.Int)
-		if err != nil {
-			return err
+		var jobID jobspb.JobID
```

Comment:
as per https://github.com/cockroachdb/cockroach/issues/108500#issuecomment-1673902483, you could consider `tree.TypeCheckandRequire`

Reply:
Good to know for the future; but I think i'd rather avoid Must* functions.

Reply:
well i'm a bit confused. it seems to do the exact thing that you asked for in https://github.com/cockroachdb/cockroach/issues/108500#issuecomment-1673854216

Reply:
Yeah, it does.  Except that this method is not on the type -- but it's a standalone function.
I didn't know it existed.  But regardless, I think not using MustBeDInt is better, in general.  I mean, every time I see these Must functions i must ask myself (pun intended) if this stuff runs under recover or not. So, let's not wonder, and just not do it.

Reply:
Might need typecheck afterall.